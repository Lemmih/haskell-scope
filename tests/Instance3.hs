module Instance3 where

data String

class Show a where
  show :: a -> String

instance String `Show` String
