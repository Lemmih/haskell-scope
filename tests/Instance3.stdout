Scope errors:

Definitions:
  Definition 1, from Instance3.decl0:
    data [4mString[0m
  Definition 2, from Instance3.decl1:
    class [4mShow[0m a where
  Definition 3, from Instance3.decl1:
    class Show [4ma[0m where
  Definition 4, from Instance3.decl1:
      [4mshow[0m :: a -> String

Use sites:
  Definition used: 3
      show :: [4ma[0m -> String
  Definition used: 1
      show :: a -> [4mString[0m
  Definition used: 1
    instance [4mString[0m `Show` String
  Definition used: 2
    instance String `[4mShow[0m` String
  Definition used: 1
    instance String `Show` [4mString[0m

Exports:
  Show, Class from Instance3
  String, Data from Instance3
  show, Method from Instance3
